### DISCUSSION TITLE
Dark mode background toggle

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [histoire-dev/histoire#215] [Q&A] [answered]
Hi folks!

Thank you so much for this library! I come from Storybook and it's great to see faster development times and a much better UI.

I'm kind of stuck at a point where I'm unable to sync the background colour of the canvas along with the dark mode of the UI. Here's my situation:

1. I've my components designed with different colours for light/dark mode. I wish to preview both of those.
2. I can toggle light/dark mode from top-left near the logo, but the background colour of the canvas remains the one selected from the top-right menu.
3. I have added transparent as an option, but since the background colour of the Histoire Sandbox cannot be changed, it doesn't show me an actual representation of how my component looks on my background colour.

For example:

This is with transparent background and dark mode enabled:

<img width="118" alt="image" src="https://user-images.githubusercontent.com/11403551/181905435-a17b714d-8f1c-410f-89d5-d124870ce940.png">

This is the actual background colour I want to see (might be hard to see on GitHub dark mode):

<img width="126" alt="image" src="https://user-images.githubusercontent.com/11403551/181905448-42b476c0-5ad4-4998-87bb-d233fb89fddf.png">

It's not intuitive to toggle canvas colour and theme separately. Is there some option that I'm missing or is this a feature request?

I'm sorry, I spent quite some time looking, but I did not find what I needed.

**Comment by victorlmneves** [comment]
@Hrishikesh-K aren't you looking for this?
<img width="349" alt="image" src="https://user-images.githubusercontent.com/17341789/182908129-68a77364-ff05-4d6a-b547-ffc801c3cd67.png">
This is where you change the canvas background color

**Reply by hrishikesh-k** [reply]
Yes, but I don't find a way to "link" that with the dark mode toggle for histoire. Why can't I just change the canvas background? Because that doesn't trigger the `dark` class name for the HTML container of my variants and thus, my CSS variables for the `.dark` selector, do not work.

To summarise, my problem is, I've my CSS variables like:

```css
:root {
  --some-var: value
}
.dark {
  --some-var: dark-value
}
```

This would work only when `<html>` has the class `dark` in its class list. The only way I've found that I can add this class is by using the toggle button in the top-left.

**Comment by Akryum** [comment, marked as answer]
You should be able to use the `.htw-dark` CSS class to apply some background yourself

**Reply by hrishikesh-k** [reply]
Ahh, the plain old CSS trick. Sorry, I should have thought of this, guess I got too much carried away in finding out JavaScript-based ways to achieve this.
